sycl/include/CL/sycl/usm/usm_allocator.hpp

```diff
@@ -87,8 +87,7 @@ class usm_allocator {
       usm::alloc AllocT = AllocKind,
       typename std::enable_if<AllocT == usm::alloc::device, int>::type = 0>
   void destroy(pointer Ptr) {
-    throw feature_not_supported(
-        "Device pointers do not support destroy on host", PI_INVALID_OPERATION);
+    // This method must be a NOP for device pointers.
```

Comment:
Why not if/def out with SYCL_DEVICE macro instead ?

Reply:
Isn't std::enable_if more idiomatic for C++11 than ifdef?

Reply:
I thought the throw being removed was due to the device compiler not parsing that code, but I see it makes no sense. Enable if makes more sense.